packages/schematics/src/command-line/affected.ts

```diff
@@ -42,41 +58,19 @@ function printError(command: string, e: any) {
   console.error(e.message);
 }
 
-function build(apps: string[], rest: string[]) {
+function build(apps: string[], rest: string[], parallel: boolean) {
   if (apps.length > 0) {
     console.log(`Building ${apps.join(', ')}`);
 
-    const parallel = yargsParser(rest, {
-      default: {
-        parallel: false
-      },
-      boolean: ['parallel']
-    }).parallel;
-
-    const buildCommands = rest.filter(a => !a.startsWith('--parallel') && !a.startsWith('--no-parallel'));
-    if (parallel) {
-      runAll(
-        apps.map(app => `ng build -- ${buildCommands.join(' ')} -a=${app}`),
-        {
-          parallel,
-          stdin: process.stdin,
-          stdout: process.stdout,
-          stderr: process.stderr
-        }
-      )
-        .then(() => console.log('Build succeeded.'))
-        .catch(err => console.error('Build failed.'));
-    } else {
-      apps.forEach(app => {
-        execSync(
-          `node ${ngPath()} build ${buildCommands.join(' ')} -a=${app}`,
-          {
-            stdio: [0, 1, 2]
-          }
-        );
-      });
-    }
-
+    const buildCommands = filterNxSpecificArgs(rest);
+    runAll(apps.map(app => `ng build ${buildCommands.join(' ')} -a=${app}`), {
```

Comment:
runAll hangs on TFS even when parallel is set to false. You need to move the if statement back.